credentials/credentials.go

```diff
@@ -55,17 +55,17 @@ var (
 	alpnProtoStr = []string{"h2-14", "h2-15", "h2-16"}
 )
 
-// Credentials defines the common interface all supported credentials must
-// implement.
-type Credentials interface {
-	// GetRequestMetadata gets the current request metadata, refreshing
+// Retriever is an interface that represents types that can
+// retrieve authentication credentials as metadata.
+type Retriever interface {
+	// Metadata gets the current request metadata, refreshing
 	// tokens if required. This should be called by the transport layer on
 	// each request, and the data should be populated in headers or other
 	// context. When supported by the underlying implementation, ctx can
 	// be used for timeout and cancellation.
 	// TODO(zhaoq): Define the set of the qualified keys instead of leaving
 	// it as an arbitrary string.
-	GetRequestMetadata(ctx context.Context) (map[string]string, error)
+	Retrieve(ctx context.Context) (map[string]string, error)
```

Comment:
On Tue, Mar 17, 2015 at 1:10 AM, Geert-Johan Riemer <
notifications@github.com> wrote:

> In credentials/credentials.go
> <https://github.com/grpc/grpc-go/pull/115#discussion_r26555073>:
>
> >  	// tokens if required. This should be called by the transport layer on
> >  	// each request, and the data should be populated in headers or other
> >  	// context. When supported by the underlying implementation, ctx can
> >  	// be used for timeout and cancellation.
> >  	// TODO(zhaoq): Define the set of the qualified keys instead of leaving
> >  	// it as an arbitrary string.
> > -	GetRequestMetadata(ctx context.Context) (map[string]string, error)
> > +	Retrieve(ctx context.Context) (map[string]string, error)
>
> nil is a value too..
>
It can return anything. But the point is that the caller uses a Retriever
but does not care whatever is retrieved from it.

> In the same way GetRequestMetadata is a getter that does not get anything?
>
yes, but the struct is NOT called something like
"RequestMetadataRequester". There is a close binding between "Retriever"
and "Retrieve()".

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/115/files#r26555073>.
>